CardScan/Classes/SSDOcrDetect.swift

```diff
@@ -0,0 +1,274 @@
+    }
+    
+    func detectOcrObjects(prediction: SSDOcrOutput, image: UIImage) -> String? {
+        var DetectedOcrBoxes = DetectedAllOcrBoxes()
+        
+        var scores : [[Float]]
+        var boxes : [[Float]]
+        var filterArray : [Float]
+
+        (scores, boxes, filterArray) = prediction.getScores(filterThreshold: filterThreshold)
+        let regularBoxes = prediction.convertLocationsToBoxes(locations: boxes,
+                                                              priors: SSDOcrDetect.priors ?? OcrPriorsGen.combinePriors(),
+                                                              centerVariance: 0.1, sizeVariance: 0.2)
+        let cornerFormBoxes = prediction.centerFormToCornerForm(regularBoxes: regularBoxes)
+        
+        var prunnedScores : [[Float]]
+        var prunnedBoxes : [[Float]]
+        
+        (prunnedScores, prunnedBoxes) = prediction.filterScoresAndBoxes(scores: scores,
+                                                                         boxes: cornerFormBoxes,
+                                                                         filterArray:  filterArray,
+                                                                         filterThreshold: filterThreshold)
+        
+        if prunnedScores.isEmpty || prunnedBoxes.isEmpty{
+            prunnedScores = [[Float]](repeating: [Float](repeating: 0.0, count: 2), count: 2)
+            prunnedBoxes = [[Float]](repeating: [Float](repeating: 0.0, count: 2 ), count: 2)
+            
+        }
+        
+        let predUtil = PredictionUtilOcr()
+        let result:Result = predUtil.predictionUtil(scores:prunnedScores, boxes: prunnedBoxes,
+                                                  probThreshold: probThreshold,
+                                                  iouThreshold: iouThreshold,
+                                                  candidateSize: candidateSize,
+                                                  topK: topK)
+    
+        for idx in 0..<result.pickedBoxes.count {
+            DetectedOcrBoxes.allBoxes.append(DetectedSSDOcrBox(category: result.pickedLabels[idx], conf: result.pickedBoxProbs[idx],
+                                                               XMin: Double(result.pickedBoxes[idx][0]), YMin: Double(result.pickedBoxes[idx][1]),
+                                                               XMax: Double(result.pickedBoxes[idx][2]), YMax: Double(result.pickedBoxes[idx][3]),
+                                                               imageSize: image.size))
+        }
+        
+        if self.isQuickRead(allBoxes: DetectedOcrBoxes){
+            let _cardNumber = processQuickRead(allBoxes: DetectedOcrBoxes)
+            return _cardNumber
+        }
+        else {
+            //os_log("%@", type: .error, "Not Quick Read")
+        }
+        
+        if (!result.pickedBoxes.isEmpty) {
+                let topCordinates = result.pickedBoxes.map{$0[1]}
```

Comment:
hard coded array indexes are risky in swift since we can't catch them if we go OOB. Can we add some unit tests here to make sure that it is safe?

Reply:
How about creating an extension to Arrays that checks if this index exists and returns nil otherwise?